script.js

```diff
@@ -1,4 +1,145 @@
+// click submit
+// player 1 roll 2 dice and show dice rolls
+var diceResultPlayer1 = [];
+var diceOrderPlayer1 = "";
+var diceResultPlayer2 = [];
+var diceOrderPlayer2 = "";
+var myOutputValue = "";
+var gameMode = `Player 1 rolls`;
+var Player1Score = 0;
+var Player2Score = 0;
+
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+var rollDiceResultPlayer1 = function () {
+  if (diceResultPlayer1 == 0) {
+    var diceRoll1 = rollDice();
+    var diceRoll2 = rollDice();
+    diceResultPlayer1.push(diceRoll1, diceRoll2);
+  } else {
+  }
+  return diceResultPlayer1;
+};
+
+// player 1 enters order of dice - 1 for same or 2 to swap, then click submit
+var decideDiceOrderPlayer1 = function (input, diceResultPlayer1) {
+  if (input == 1 && gameMode == "Player 1 rolls") {
+    diceOrderPlayer1 = "" + diceResultPlayer1[0] + diceResultPlayer1[1];
+  } else if (input == 2 && gameMode == "Player 1 rolls") {
+    diceOrderPlayer1 = "" + diceResultPlayer1[1] + diceResultPlayer1[0];
+  }
+
+  return Number(diceOrderPlayer1);
+};
+
+// player 2 roll 2 dice and show dice rolls
+var rollDiceResultPlayer2 = function () {
+  if (diceResultPlayer2 == 0) {
+    var diceRoll3 = rollDice();
+    var diceRoll4 = rollDice();
+    diceResultPlayer2.push(diceRoll3, diceRoll4);
+  } else {
+  }
+  return diceResultPlayer2;
+};
+
+// player 2 enters order of dice - 1 for same or 2 to swap, then click submit
+var decideDiceOrderPlayer2 = function (input, diceResultPlayer2) {
+  if (input == 1 && gameMode == "Player 2 rolls") {
+    diceOrderPlayer2 = Number("" + diceResultPlayer2[0] + diceResultPlayer2[1]);
+  } else if (input == 2 && gameMode == "Player 2 rolls") {
+    diceOrderPlayer2 = Number("" + diceResultPlayer2[1] + diceResultPlayer2[0]);
+  }
+  return diceOrderPlayer2;
+};
+
+//compare numbers, player with higher number wins
+var decideWinner = function (input, diceOrderPlayer1, diceOrderPlayer2) {
+  var diceOrderPlayer1 = decideDiceOrderPlayer1(input, diceResultPlayer1);
+  var diceOrderPlayer2 = decideDiceOrderPlayer2(input, diceResultPlayer2);
+  var winner = `It's a draw!`;
+  if (diceOrderPlayer1 > diceOrderPlayer2) {
+    winner = `Player 1 wins!`;
+  } else if (diceOrderPlayer2 > diceOrderPlayer1) {
+    winner = `Player 2 wins!`;
+  }
+  return winner;
+};
+
+// valid inputs are "", 1 or 2.
+var checkInputValid = function (input) {
+  inputValid = false;
+  if (
+    (input == "" && gameMode == `Player 1 rolls`) ||
+    input == 1 ||
+    input == 2
+  ) {
+    inputValid = true;
+  }
+  return inputValid;
+};
```

Comment:
I suppose this function, once completed & implemented, will restrict the users in their inputs. Another way to force the user to input an intended value is through game modes (i.e., only after user inputs an intended value, would the game mode change).